cmd/syft/cli/options/file.go

```diff
@@ -22,6 +23,10 @@ type fileContent struct {
 	Globs              []string `yaml:"globs" json:"globs" mapstructure:"globs"`
 }
 
+type fileExecutable struct {
+	Globs []string `yaml:"globs" json:"globs" mapstructure:"globs"`
```

Comment:
open question: does this approach make sense to folks? Right now this allows for sub-selecting by path glob, and keys off of the file.Selection to do nothing when there is no selection specified. I think it's a little awkward since:
- this cataloger implies that only binaries are scanned
- it does not yet account for the owned-files selection, which I think should be added to this PR.